SteamcLog/Classes/SteamcLog.swift

```diff
@@ -245,6 +247,21 @@ public struct SteamcLog {
         warn("\(message): \(jsonString)", functionName: functionName, fileName: fileName, lineNumber: lineNumber)
     }
 
+    @_disfavoredOverload
+    public func warn<T>(_ message: String, _ error: T, functionName: StaticString = #function, fileName: StaticString = #file, lineNumber: Int = #line) where T: Error {
+        if config.requireRedacted {
```

Comment:
would this be clearer as a `guard`?

Reply:
Yeah, I'm a little torn on that. They do need to early return (and it might be nice to enforce that), the things that make me not want to do it are:
- All of the functions are like that, I just copied those from the existing code for the other forms of this, so just fixing the new one would make them inconsistent, we'd probably want to change all of them.
- `if requireRedacted` reads clearer to me about what case is actual going to what branch then `guard !requireRedacted` 
- If we were gonna change all of them, I think there are actually better ways to approach this, like rather than having a bunch of different copies of these checks in each function just factor out one function to generate the text string so that the individual functions don't have any early outs. 

I'm inclined to leave it for now, but fill a bug to refactor that.

Reply:
Filed https://github.com/steamclock/steamclog-swift/issues/106